executor/admin.go

```diff
@@ -0,0 +1,157 @@
+// Copyright 2018 PingCAP, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package executor
+
+import (
+	"github.com/juju/errors"
+	"github.com/pingcap/tidb/ast"
+	"github.com/pingcap/tidb/distsql"
+	"github.com/pingcap/tidb/infoschema"
+	"github.com/pingcap/tidb/model"
+	"github.com/pingcap/tidb/plan"
+	"github.com/pingcap/tidb/table"
+	"github.com/pingcap/tidb/types"
+	"github.com/pingcap/tidb/util/chunk"
+	"github.com/pingcap/tidb/util/ranger"
+	"github.com/pingcap/tipb/go-tipb"
+	"golang.org/x/net/context"
+)
+
+var _ Executor = new(CheckIndexRangeExec)
+
+// CheckIndexRangeExec output the index values which has handle between begin and end.
+type CheckIndexRangeExec struct {
+	baseExecutor
+
+	table    table.Table
+	index    table.Index
+	is       infoschema.InfoSchema
+	startKey []types.Datum
+
+	handleRanges []ast.HandleRange
+	srcChunk     *chunk.Chunk
+
+	result   distsql.SelectResult
+	partial  distsql.PartialResult
+	typesMap map[int64]*types.FieldType
+	cols     []*model.ColumnInfo
+}
+
+// NextChunk implements the Executor Next interface.
+func (e *CheckIndexRangeExec) NextChunk(ctx context.Context, chk *chunk.Chunk) error {
+	chk.Reset()
+	handleIdx := e.schema.Len() - 1
+	for {
+		err := e.result.NextChunk(ctx, e.srcChunk)
+		if err != nil {
+			return errors.Trace(err)
+		}
+		if e.srcChunk.NumRows() == 0 {
+			return nil
+		}
+		for i := 0; i < e.srcChunk.NumRows(); i++ {
```

Comment:
```go
iter := chunk.NewIterator4Chunk(e.srcChunk)
for row := iter.Begin(); row != iter.End(); row = iter.Next() {
}
```

and `iter := chunk.NewIterator4Chunk(e.srcChunk)` can be defined outside `for` loop